### PR TITLE
Add option to calculate formal molecular charge to rdkit descriptor tool

### PR DESCRIPTION
**Pull request by simonbray**
Also, lint python files in the rdkit dir

### CODE REVIEW

File: chemicaltoolbox/rdkit/rdkit_get_charge.xml
Comment (bgruening) [inline chemicaltoolbox/rdkit/rdkit_get_charge.xml:1]: from a user-point of view, should this be added to https://github.com/bgruening/galaxytools/blob/master/chemicaltoolbox/openbabel/ob_genProp.xml --> even if we cross the rdkit/openbabel boundary here (which is purely technical and does not make much sense to the user maybe?)

Comment (simonbray) [reply chemicaltoolbox/rdkit/rdkit_get_charge.xml:1]: I guess we could but I'm not so enthusiastic. I think our users are quite well informed about different packages and it should be clear from the tool which one they are using.

Or is your concern generally about creating a lot of tiny tools with very specific functionality?

Comment (bgruening) [reply chemicaltoolbox/rdkit/rdkit_get_charge.xml:1]: More the latter. More tools means more maintenance and more things that can break.

We could still offer them the information (charge (ob) vs. charge (rdkit)) but having two tools for charge calculation, just with a different implementation seems weird.

But I understand that this is not an important topic. So I'm happy to merge that if you prefer.

Comment (simonbray) [reply chemicaltoolbox/rdkit/rdkit_get_charge.xml:1]: There is also the rdkit descriptor tool. I can add it to that instead.

> We could still offer them the information (charge (ob) vs. charge (rdkit)) but having two tools for charge calculation, just with a different implementation seems weird.

I don't think charge can be calculated with the openbabel tool?

Comment (bgruening) [reply chemicaltoolbox/rdkit/rdkit_get_charge.xml:1]: > There is also the rdkit descriptor tool. I can add it to that instead.

Also good! You see we have already so many places. If you concerned about performance, because to many things are calculated we can make those optional. 

> I don't think charge can be calculated with the openbabel tool?

No it can't to my knowledge, it was just an example.

**Comment by simonbray** [comment]
do not merge yet, I need to test again tomorrow

File: chemicaltoolbox/rdkit/rdkit_descriptors.py
Comment (bgruening) [inline chemicaltoolbox/rdkit/rdkit_descriptors.py:9]: We still have a problem with pulsar that those extra files (get_supplier.py) are not moved to the pulsar endpoint if they are not specified/called in the XML file.

So either you include (touch $__tool_directory__/get_supplier.py) in the command section or you leave this function in this main file.

File: chemicaltoolbox/rdkit/rdkit_descriptors.xml
Comment (bgruening) [inline chemicaltoolbox/rdkit/rdkit_descriptors.xml:17]: Will this be None in case of blank? `#if` might be more reliable?

File: chemicaltoolbox/rdkit/rdkit_descriptors.py
Comment (simonbray) [reply chemicaltoolbox/rdkit/rdkit_descriptors.py:9]: Ah, good to know

**Review by bgruening — APPROVED** [review]
Good from my side! Thanks!

**Pull request merged**
